dvc/repo/imp_url.py

```diff
@@ -27,6 +29,9 @@ def imp_url(
     out = resolve_output(url, out)
     path, wdir, out = resolve_paths(self, out)
 
+    if to_remote and no_exec:
+        raise ValueError("to-remote option can't be combined with no_exec")
```

Comment:
We tend to use `InvalidArgumentError` for this. You can also call the options by their CLI names and then you won't really need same checks in CLI. E.g. see dvc/repo/run.py.

Reply:
Ah, I see. Thanks!